src/main/scala/chisel3/util/SRAM.scala

```diff
@@ -106,6 +107,34 @@ class SRAMInterface[T <: Data](
 
 object SRAM {
 
+  /** Generates a [[SyncReadMem]] within the current module, connected to an explicit number
+    * of read, write, and read/write ports. This SRAM abstraction has both read and write capabilities: that is,
+    * it contains at least one read accessor (a read-only or read-write port), and at least one write accessor
+    * (a write-only or read-write port).
+    *
+    * @param size The desired size of the inner `SyncReadMem`
+    * @tparam T The data type of the memory element
+    * @param numReadPorts The number of desired read ports >= 0, and (numReadPorts + numReadwritePorts) > 0
+    * @param numWritePorts The number of desired write ports >= 0, and (numWritePorts + numReadwritePorts) > 0
+    * @param numReadwritePorts The number of desired read/write ports >= 0, and the above two conditions must hold
+    * @param contents A filesystem path to a binary file to preload this SRAM's contents with
+    *
+    * @return A new `SRAMInterface` wire containing the control signals for each instantiated port
+    * @note This does *not* return the `SyncReadMem` itself, you must interact with it using the returned bundle
+    * @note Read-only memories (R >= 1, W === 0, RW === 0) and write-only memories (R === 0, W >= 1, RW === 0) are not supported by this API, and will result in an error if declared.
+    */
+  def apply[T <: Data](
+    size:              BigInt,
+    tpe:               T,
+    numReadPorts:      Int,
+    numWritePorts:     Int,
+    numReadwritePorts: Int,
+    contents:          Option[String] = None
```

Comment:
```suggestion
    loadMemoryFile:          String
```

The name `contents` is a bit misleading to me, I think making it more obvious that this is a filename is helpful.

Reply:
Also, this should not have a default value because otherwise it appears to be ambiguous with the version of `apply` that does not have this extra argument. I'm surprised the compiler doesn't complain about it, but in any case, the version where `contents` is not set is just the other version of this method.